File: app/src/main/java/com/example/pagingtmdbdapr/activity/Coreframework.java
package com.example.pagingtmdbdapr.activity;

import android.content.Context;

class Coreframework {
  static   Context context;
    static Coreframework pagingSource;
    Coreframework()
    {

    }
    public static Coreframework getInstance(Context mC) {
        if (context == null) {
context = mC;
        }

        if (pagingSource == null){
            pagingSource = new Coreframework();
        }
        return  pagingSource;
    }
}

File: app/src/main/java/com/example/pagingtmdbdapr/util/Coreframework.kt
package com.example.pagingtmdbdapr.util

import android.content.Context

class Coreframework {
    var context: Context? = null
    var pagingSource: Coreframework? = null
    fun getInstance(mC: Context?): Coreframework? {
        if (context == null) {
            context = mC
        }
        if (pagingSource == null) {
            pagingSource = Coreframework()
        }
        return pagingSource
    }
}